exporter/elasticsearchexporter/bulkindexer.go

```diff
@@ -126,8 +126,12 @@ type syncBulkIndexerSession struct {
 }
 
 // Add adds an item to the sync bulk indexer session.
-func (s *syncBulkIndexerSession) Add(ctx context.Context, index string, document io.WriterTo, dynamicTemplates map[string]string) error {
-	err := s.bi.Add(docappender.BulkIndexerItem{Index: index, Body: document, DynamicTemplates: dynamicTemplates})
+func (s *syncBulkIndexerSession) Add(ctx context.Context, index string, docID string, document io.WriterTo, dynamicTemplates map[string]string) error {
+	doc := docappender.BulkIndexerItem{Index: index, Body: document, DynamicTemplates: dynamicTemplates}
+	if docID != "" {
+		doc.DocumentID = docID
+	}
```

Comment:
```suggestion
	doc := docappender.BulkIndexerItem{
		Index: index,
		Body: document,
		DocumentID: docID,
		DynamicTemplates: dynamicTemplates
	}
```

It doesn't need to be conditional: the field is a string, and it'll be omitted from the encoded JSON if it's empty.

Reply:
Fixed, thanks!